From b05dd7637478147a8ec42f79ea163d5c7983b227 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sun, 3 Jan 2021 18:18:34 +0100
Subject: [PATCH 3/3] Add option to disable data generation

The data generation is not needed when only the library and the
utilities are built.
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 467c90c..9a3ee9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,7 @@ find_package(GLIB2 REQUIRED)
 
 option(INSTALL_UTILS "Install generator utilities" OFF)
 option(DOWNLOAD_MODEL_DATA_ARCHIVE "Update the model data archive while building" ON)
+option(DISABLE_DATA_GENERATION OFF)
 
 # DBM: BerkeleyDB
 find_package(BerkeleyDB)
@@ -185,4 +186,6 @@ include_directories(
 add_subdirectory(src)
 add_subdirectory(tests)
 add_subdirectory(utils)
-add_subdirectory(data)
+if(NOT DISABLE_DATA_GENERATION)
+    add_subdirectory(data)
+endif()
-- 
2.27.0

